store/store.go

```diff
@@ -915,16 +914,11 @@ func (s *Store) WaitForFSMIndex(idx uint64, timeout time.Duration) (uint64, erro
 	defer tck.Stop()
 	tmr := time.NewTimer(timeout)
 	defer tmr.Stop()
-
-	var fsmIdx uint64
 	for {
 		select {
 		case <-tck.C:
-			s.fsmIdxMu.RLock()
-			fsmIdx = s.fsmIdx
-			s.fsmIdxMu.RUnlock()
-			if fsmIdx >= idx {
-				return fsmIdx, nil
+			if s.fsmIdx.Load() >= idx {
+				return s.fsmIdx.Load(), nil
```

Comment:
The first and second Load() isn't synchronized. I assume in this context it's ok if the returned value is different than the one compared?  The original mutex-based code this same behavior too so I imagine it's fine, but I figure it can't hurt to double check.

Reply:
Yeah, it always goes up so it doesn't matter, but actually there is a neater way to fix. Thanks for flagging.

Reply:
https://github.com/rqlite/rqlite/commit/248c66f251cbea10094c2074c6623a7955deb63f